pkg/sql/logictest/testdata/logic_test/fk

```diff
@@ -874,15 +874,22 @@ pkref_b  CREATE TABLE pkref_b (
 
 # See https://github.com/cockroachdb/cockroach/issues/20045
 statement ok
-CREATE TABLE self_x2 (
+CREATE TABLE test20045 (
   x STRING PRIMARY KEY
- ,y STRING UNIQUE REFERENCES self_x2(x)
- ,z STRING REFERENCES self_x2(y)
- ,INDEX(z)
+ ,y STRING UNIQUE REFERENCES test20045(x)
+ ,z STRING REFERENCES test20045(y)
 );
 
 statement ok
-INSERT INTO self_x2 (x, y, z) VALUES ('pk1', NULL, NULL);
+INSERT INTO test20045 (x, y, z) VALUES ('pk1', NULL, NULL);
 
 statement ok
-DELETE FROM self_x2 WHERE x = 'pk1';
+DELETE FROM test20045 WHERE x = 'pk1';
+
+# See https://github.com/cockroachdb/cockroach/issues/20042
+statement ok
+CREATE TABLE test20042 (
+  x STRING PRIMARY KEY
+ ,y STRING UNIQUE
+ ,z STRING REFERENCES test20042(y)
+);
```

Comment:
Wanna add an INSERT and DELETE after this to make sure that it works for this schema